internal/provider/resource_time_rotating_test.go

```diff
@@ -66,369 +70,1224 @@ func TestAccTimeRotating_Triggers(t *testing.T) {
 	})
 }
 
-func TestAccTimeRotating_RotationDays_basic(t *testing.T) {
+func TestAccTimeRotating_ComputedRFC3339RotationDays_basic(t *testing.T) {
 	resourceName := "time_rotating.test"
-	timestamp := time.Now().UTC()
+
+	now := time.Now().UTC()
+	mockClock := fakeclock.NewFakeClock(now)
+	rotationDays := 7
 
 	resource.UnitTest(t, resource.TestCase{
-		ProtoV5ProviderFactories: protoV5ProviderFactories(),
-		CheckDestroy:             nil,
+		CheckDestroy: nil,
 		Steps: []resource.TestStep{
 			{
-				Config: testAccConfigTimeRotatingRotationDays(timestamp.Format(time.RFC3339), 7),
+				ProtoV5ProviderFactories: protoV5ProviderFactoriesTestProvider(mockClock),
+				Config:                   testAccConfigTimeRotatingRotationDays(rotationDays),
+				ConfigPlanChecks: resource.ConfigPlanChecks{
+					PreApply: []plancheck.PlanCheck{
+						plancheck.ExpectResourceAction(resourceName, plancheck.ResourceActionCreate),
+						plancheck.ExpectUnknownValue(resourceName, tfjsonpath.New("rotation_rfc3339")),
+						plancheck.ExpectUnknownValue(resourceName, tfjsonpath.New("rfc3339")),
+						plancheck.ExpectKnownValue(resourceName, tfjsonpath.New("rotation_days"), knownvalue.Int64Exact(int64(rotationDays))),
+					},
+				},
 				ConfigStateChecks: []statecheck.StateCheck{
-					statecheck.ExpectKnownValue(resourceName, tfjsonpath.New("rotation_days"), knownvalue.Int64Exact(7)),
-					statecheck.ExpectKnownValue(resourceName, tfjsonpath.New("rotation_rfc3339"), knownvalue.StringExact(timestamp.AddDate(0, 0, 7).Format(time.RFC3339))),
-					statecheck.ExpectKnownValue(resourceName, tfjsonpath.New("rotation_years"), knownvalue.Null()),
-					statecheck.ExpectKnownValue(resourceName, tfjsonpath.New("rotation_months"), knownvalue.Null()),
-					statecheck.ExpectKnownValue(resourceName, tfjsonpath.New("rotation_hours"), knownvalue.Null()),
-					statecheck.ExpectKnownValue(resourceName, tfjsonpath.New("rotation_minutes"), knownvalue.Null()),
```

Comment:
I've removed these null assertions from most of the tests, since these are configured attributes and they clutter up the test assertions.